### PR TITLE
Support ImmutableJS Map as root state object

### PR DESCRIPTION
**Pull request by whatisboom**
Updated functions per our conversation to add support for ImmutableJS Maps as root state object

### CODE REVIEW

File: src/util/lastUpdate.js [hunk 1]
```diff
@@ -28,8 +28,11 @@ export const getLastUpdate = (store, key, reducerKeys = REDUCER_KEYS) => {
 
     return keys.reduce((prev, reducerAccessor) => {
         const reducerKey = reducerKeys[reducerAccessor];
-        if (state[reducerKey] && state[reducerKey].toJS) {
-            prev[reducerKey] = state[reducerKey].getIn([key, 'lastUpdate']);
+        const stateMap = (typeof state.get === 'function') ?
```
Comment (bencripps) [inline src/util/lastUpdate.js]: it's a nit, but could you update the syntax to the following

const stateMap = (typeof state.get === 'function') 
    ? state.get(reducerKey)
    : state[reducerKey];

**Comment by bencripps** [comment]
Changes look good; let's confirm it fixes all of your issues before the MR.

Can you squash both (and any following commits) into a single commit?

I will add some test cases around the changes once the MR is complete.

**Comment by bencripps** [comment]
Just as a general comment, this MR satisfies the case where the `redux` state atom is initialized as an *Immutable* `map`, instead of a plain JavaScript object. This does not alter the way the grid component functions, and although the grid internally uses *Immutable* data structures, that is an implementation detail that isn't exposed through the public API.

TLDR; by merging this MR, the component will be able to write and read state from the state atom as an object or an Immutable `map`.
